lib/ex_aws/s3.ex

```diff
@@ -1006,15 +1006,16 @@ defmodule ExAws.S3 do
     {hash_header(alg), :crypto.hash(alg, content) |> Base.encode64()}
   end
 
-  @spec get_hash_config() :: :md5
+  @spec get_hash_config() :: atom()
   defp get_hash_config() do
     Application.get_env(:ex_aws_s3, :content_hash_algorithm) || :md5
   end
 
   # Supported hash algorithms:
   # https://www.erlang.org/doc/man/crypto.html#type-hash_algorithm
   @spec hash_header(atom()) :: binary()
-  defp hash_header(alg) when is_atom(alg), do: "content-#{to_string(alg)}"
+  defp hash_header(:md5), do: "content-md5"
+  defp hash_header(alg) when is_atom(alg), do: "x-amz-checksum-#{to_string(alg)}"
```

Comment:
Nice, I noticed the "spec" seems to be using `x-amz-checksum`, seems like this will be a good addition